src/NexusMods.DurableJobs/JobManager.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System.Collections.Concurrent;
+using System.Text.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.DurableJobs;
+
+namespace NexusMods.DurableJobs;
+
+/// <summary>
+/// Implementation of the job manager.
+/// </summary>
+public class JobManager : IJobManager, IHostedService
+{
+    private readonly ConcurrentDictionary<JobId, IActor<IJobMessage>> _jobs = new();
+    private readonly ILogger<JobManager> _logger;
+    private readonly JsonSerializerOptions _serializerOptions;
+    private readonly Dictionary<Type,IJob> _jobInstances;
+    private readonly IJobStateStore _jobStore;
+
+    /// <summary>
+    /// DI constructor.
+    /// </summary>
+    public JobManager(IServiceProvider serviceProvider)
+    {
+        _jobStore = serviceProvider.GetRequiredService<IJobStateStore>();
+        _jobInstances = serviceProvider.GetServices<AOrchestration>().OfType<IJob>()
+            .Concat(serviceProvider.GetServices<AUnitOfWork>())
+            .ToDictionary(j => j.GetType());
+        _logger = serviceProvider.GetRequiredService<ILogger<JobManager>>();
+        _serializerOptions = serviceProvider.GetRequiredService<JsonSerializerOptions>();
+    }
+    
+    
+    
+    /// <inheritdoc />
+    public Task<object> RunNew<TJob>(params object[] args)
+        where TJob : AOrchestration
+    {
+        var jobInstance = _jobInstances[typeof(TJob)];
+        var newJobId = JobId.From(Guid.NewGuid());
+        
+        var tcs = new TaskCompletionSource<object>();
+
+        var initialState = new OrchestrationState
+        {
+            Job = (AOrchestration)jobInstance,
+            Id = newJobId,
+            Manager = this,
+            Arguments = args,
+            Continuation = Continuation,
+        };
+        
+        var actor = new JobActor(_logger, initialState);
+        actor.Post(RunMessage.Instance);
+        _jobs[newJobId] = actor;
+        
+        return tcs.Task;
+        
+
+        void Continuation(object obj, Exception? ex)
+        {
+            if (ex != null)
+            {
+                tcs.TrySetException(ex);
+            }
+            else
+            {
+                tcs.TrySetResult(obj);
+            }
+        }
+    }
+    
+    /// <summary>
+    /// Cancel the given job, this will trigger a cancel message to be sent to all of its children and parents.
+    /// </summary>
+    internal void CancelJob(JobId stateParentJobId)
+    {
+        if (!_jobs.TryGetValue(stateParentJobId, out var parentJob))
+            return;
+        parentJob.Post(CancelMessage.Instance);
+    }
+
+    internal AJobState SaveState(AJobState state)
```

Comment:
The idea is that in debug mode we're constantly serialzing and deserialzing the state so that we can make sure the user didn't forget at some point to add serializers and end up with a job in release mode that can't be deserialized.